### PR TITLE
Update `IMX` inference wrapper

### PR DESCRIPTION
**Pull request by Laughing-q**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Improves IMX export/inference consistency by simplifying the decode path and ensuring anchors/strides refresh correctly for changing input shapes 🧩🚀

### 📊 Key Changes
- Updated YOLO head anchor/stride refresh logic to **always recompute** when running in dynamic mode or when the feature-map shape changes (removed the special-case skip for `format="imx"`) ⚙️
- Simplified IMX `_inference` to **only accept dict inputs** (`{"boxes": ..., "scores": ...}`), removing the alternative list-of-tensors concatenation path 🧹
- IMX decoding now directly uses `x["boxes"]` and `x["scores"]` (with sigmoid applied to scores), making the pipeline more explicit and consistent ✅

### 🎯 Purpose & Impact
- More reliable IMX behavior when input resolutions change (anchors/strides won’t become stale), reducing shape-related inference/export issues 📐🔒
- Clearer, more maintainable IMX inference code by enforcing a single input format, which can reduce ambiguity and bugs 🛠️
- Potential breaking change for any custom IMX usage that previously passed a `list[torch.Tensor]` into `_inference`—those callers must now provide the dict format 🧯

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Laughing-q, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message—an Ultralytics engineer will assist shortly. To ensure a seamless integration of your work, please review the following checklist:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/utils/export/imx.py [hunk 1]
```diff
@@ -106,13 +106,8 @@ def forward(self, x):
...
 def _inference(self, x: list[torch.Tensor] | dict[str, torch.Tensor]) -> tuple[torch.Tensor]:
     """Decode boxes and cls scores for imx object detection."""
-    if isinstance(x, dict):
-        box, cls = x["boxes"], x["scores"]
-    else:
-        x_cat = torch.cat([xi.view(x[0].shape[0], self.no, -1) for xi in x], 2)
-        box, cls = x_cat.split((self.reg_max * 4, self.nc), 1)
-    dbox = self.decode_bboxes(self.dfl(box), self.anchors.unsqueeze(0)) * self.strides
-    return dbox.transpose(1, 2), cls.sigmoid().permute(0, 2, 1)
+    dbox = self.decode_bboxes(self.dfl(x["boxes"]), self.anchors.unsqueeze(0)) * self.strides
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/export/imx.py:109]: ⚠️ **HIGH**: `_inference()` now unconditionally indexes `x["boxes"]` / `x["scores"]`, but the type hint (and likely some call sites) still allow `list[Tensor]`. If a list is passed, this will raise at runtime with a non-obvious error. Either tighten the accepted type to `dict[str, Tensor]` only, or add an explicit guard that raises a clear `TypeError` describing the required dict keys (`boxes`, `scores`).

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, focused change overall: removing the IMX special-case in anchor/stride refresh improves correctness for dynamic shapes. The main issue is robustness in `imx.py`: the decoding path now assumes dict inputs but the function signature still allows list inputs, which can cause confusing runtime failures. Tightening the type or adding a clear guard would make the breaking change safer and easier to debug.

💬 Posted 1 inline comment

**Review by lmycross — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged—this is a fantastic improvement. Thank you @Laughing-q (and @lmycross for the contributions) for pushing IMX export/inference toward a cleaner, more reliable path.

As Leonardo da Vinci said, “Simplicity is the ultimate sophistication.” This PR embodies that: by enforcing a single dict-based IMX `_inference` input and making anchor/stride refresh deterministic for dynamic shapes, you’ve reduced ambiguity and eliminated a whole class of shape-staleness pitfalls. The result is a more explicit decode pipeline and more dependable behavior when resolutions change—exactly the kind of robustness that keeps Ultralytics workflows smooth for the community.

Really appreciate the thoughtful cleanup and the focus on long-term maintainability.
